### PR TITLE
Header and Footer Layout

### LINKED ISSUE #36: Move Header and Footer to Layout.jsx
Add Header and Footer above and below {children} respectively.

DoD:

- Header renders above page content on all pages
- Footer renders below page content on all pages
- isDarkMode state is provided to header and footer components and theme toggle works

### PR DESCRIPTION
**Pull request by zijunw04**
I tried my best to implement the footer and header layout as a "theme" in react. It works perfectly fine for me syncing with all other pages along with inverting the header when user are on the "contributor" page.

### CODE REVIEW

**Comment by zijunw04** [comment]
Updated 
- Changed the layout of contributor page instead of inverting header
- Added hero to the contributor page
- Revamp the contributor box, still kept the border color that user picked before
- Dark mode is synced among different pages 
- Added 1 new animation (typewriter)

File: app/Contributors/contributorhero.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+export default function ContributorHeroScreen( {scrollToProductList, isDarkMode}) {
+  return (
+    <div style={{ background: isDarkMode?"linear-gradient(180deg, #D90000 91.67%, #262425 100%)":"#34A3D9"}} className="flex flex-col items-center  h-screen bg-cover bg-center w-full">
+      <div className="flex flex-col items-center  h-[750px] bg-cover bg-center w-full justify-center">
+      <h1 className="text-white text-9xl max-w-[970px]  font-extrabold mb-8 mt-48 text-shadow-hero overflow-hidden whitespace-nowrap shadow-HEROSHADOW animate-typewriter border-r-[.15em] ">Website Credit</h1>
+      <button style={{ color: isDarkMode ? '#D90000' : '#34A3D9' }} className="bg-white px-5 py-6 rounded-full " onClick={scrollToProductList}>
+        <a className={`text-center font-bold text-4xl text-shadow-hero shadow-SHOPTEXTSHADOW mb-[50px]`}>READ MORE</a>
+      </button>
+      </div>
+      <div className="items-center flex flex-row bg-transparent w-full">
```
Comment (Ajknight121) [inline app/Contributors/contributorhero.jsx]: Center the social media links

**Review by Ajknight121 — CHANGES_REQUESTED** [review]
Incredible work, I like that you created the useTheme function. I also really like the animation and use of svgs, just center the social media links for the contributor page and this can be pulled. If you can get the cursor to disappear at the end of the animation it could be applied to the homepage in a separate pull.

**Comment by zijunw04** [comment]
the social media should be centered and the cursor should now disappear after onAnimationEnd={}

**Review by Ajknight121 — APPROVED** [review]

**Pull request merged**